src/Managed.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {UUPSUpgradeable} from "openzeppelin-contracts/proxy/utils/UUPSUpgradeable.sol";
+import {ERC1967Proxy} from "openzeppelin-contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {StorageSlot} from "openzeppelin-contracts/utils/StorageSlot.sol";
+
+/// @notice A mix-in for contract UUPS-upgradability, pausability and admin management.
+/// It can't be used directly, only via a proxy. It uses the upgrade-safe ERC-1967 storage scheme.
+///
+/// Managed uses the ERC-1967 admin slot to store the admin address.
+/// All instances of the contracts are owned by address `0x00`.
+/// While this contract is capable of updating the admin,
+/// the proxy is expected to set up the initial value of the ERC-1967 admin.
+///
+/// All instances of the contracts are paused and can't be unpaused.
+/// When a proxy uses such contract via delegation, it's initially unpaused.
+abstract contract Managed is UUPSUpgradeable {
+    /// @notice The ERC-1967 name of the `pausedSlot` storage slot.
+    string private constant PAUSED_SLOT_NAME = "eip1967.managed.paused";
+    /// @notice The pointer to the storage slot with the boolean holding the paused state.
+    bytes32 private immutable pausedSlot;
+
+    /// @notice Emitted when the pause is triggered.
+    /// @param account The account which triggered the change.
+    event Paused(address account);
+
+    /// @notice Emitted when the pause is lifted.
+    /// @param account The account which triggered the change.
+    event Unpaused(address account);
+
+    /// @notice Initializes the contract in paused state and with no admin.
+    /// The contract instance can be used only as a call delegation target for a proxy.
+    constructor() {
+        bytes32 pausedSlot_ = erc1967Slot(PAUSED_SLOT_NAME);
+        pausedSlot = pausedSlot_;
+        StorageSlot.getBooleanSlot(pausedSlot_).value = true;
+    }
+
+    /// @notice Calculates the ERC-1967 slot pointer.
+    /// @param name The name of the slot, should be globally unique
+    /// @return slot The slot pointer
+    function erc1967Slot(string memory name) internal pure returns (bytes32 slot) {
+        return bytes32(uint256(keccak256(bytes(name))) - 1);
+    }
+
+    /// @notice Authorizes the contract upgrade. See `UUPSUpgradable` docs for more details.
+    function _authorizeUpgrade(address newImplementation) internal view override onlyAdmin {
+        newImplementation;
+    }
+
+    /// @notice Returns the address of the current admin.
+    function admin() public view returns (address) {
+        return _getAdmin();
+    }
+
+    /// @notice Changes the admin of the contract.
+    /// Can only be called by the current admin.
+    function changeAdmin(address newAdmin) public onlyAdmin {
+        _changeAdmin(newAdmin);
+    }
+
+    /// @notice Throws if called by any account other than the admin.
+    modifier onlyAdmin() {
```

Comment:
Maybe we can move the `modifier` definitions after the constructor before the regular functions.

Reply:
Fixed